api/src/pcapi/core/offers/api.py

```diff
@@ -502,9 +522,16 @@ def edit_stock(
         validation.check_booking_limit_datetime(stock, changed_beginning, changed_booking_limit)
         validation.check_required_dates_for_stock(stock.offer, changed_beginning, changed_booking_limit)
 
-    if price is not UNCHANGED and price != stock.price:
+    if price not in (UNCHANGED, None) and price != stock.price:
```

Comment:
```suggestion
    if price is not UNCHANGED and price is not None and price != stock.price:
```
je suis d'accord que c'est moins élégant mais mypy est content comme ça et tu peux retirer des `ignore` plus bas